src/main/java/uk/gov/companieshouse/itemhandler/service/FilingHistoryDescriptionProviderService.java

```diff
@@ -0,0 +1,122 @@
+            final Map<String, Object> filingHistoryDescriptionsRoot = yaml.load(inputStream);
+            filingHistoryDescriptionsLocal =
+                    (Map<String, String>) filingHistoryDescriptionsRoot.get(FILING_HISTORY_DESCRIPTION_KEY);
+            if (filingHistoryDescriptionsLocal == null) {
+                Map<String, Object> logMap = new HashMap<>();
+                logMap.put(DESCRIPTION_LOG_KEY, FILING_HISTORY_DESCRIPTION_KEY);
+                LoggingUtils.getLogger().error("Filing History descriptions file not found", logMap);
+                return null;
+            }
+        } catch (IOException ioe) {
+            // This is very unlikely to happen here given File.exists() check above,
+            // and that it is not likely to encounter an error closing the stream either.
+            LoggingUtils.getLogger().error(ioe);
+        }
+        return filingHistoryDescriptionsLocal;
+    }
+
+    private String getFilingHistoryDescriptionWithKey(String descriptionKey) {
+        if(descriptionKey == null || descriptionKey.isEmpty()) {
+            return descriptionKey;
+        } else {
+            return filingHistoryDescriptions.get(descriptionKey);
+        }
+    }
+
+    /**
+     * Fetches the specified key from the filing history enumerations and replaces the variables
+     * with the ones in the description values
+     * @param descriptionKey the key to fetch from the filing history enumerations
+     * @param filingHistoryDescriptionValues the map of the description values to replace in the description
+     * @return the formatted filing history description
+     */
+    public String mapFilingHistoryDescription(String descriptionKey, Map<String, Object> filingHistoryDescriptionValues) {
+        if(filingHistoryDescriptions == null) {
+            return null;
+        }
+        String description = getFilingHistoryDescriptionWithKey(descriptionKey);
+        if(filingHistoryDescriptionValues.containsKey("description")) {
+            return (String) filingHistoryDescriptionValues.get("description");
+        } else {
+            final StringBuilder sb = new StringBuilder(description);
+            replaceAll(sb, "*", ""); // remove all asterisks in fetched string
+            filingHistoryDescriptionValues.forEach((k,v)-> {
+                String value = k.contains("date") ? reformatActionDate((String)v) : (String) v;
+                replaceAll(sb, "{"+k+"}", value);
+            });
+            return sb.toString();
+        }
+    }
+
+    /**
+     * Reformats a date filed string such as "2009-08-23" as "23 August 2009".
+     * @param dateFiled the date filed as reported from the filing history
+     * @return the same date rendered for display purposes
+     */
+    public static String reformatActionDate(final String dateFiled) {
+        final LocalDate parsedDate = LocalDate.parse(dateFiled);
+        return parsedDate.format(DateTimeFormatter.ofPattern("dd MMMM yyyy"));
```

Comment:
This now needs to use the new `FILING_HISTORY_DATE_DESCRIPTION_FORMATTER` constant.

Reply:
I completely missed this, I'll add it in now